### PR TITLE
Fix for #828: Embed build tags

### PR DESCRIPTION
**Pull request by cez81**
Add build tags to ldflags and print in version output

### CODE REVIEW

**Comment by appleboy** [comment]
👎 Version already include tag string.

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by lunny** [comment]
seems great

**Comment by lunny** [comment]
will fix #828

File: main.go [hunk 1]
```diff
@@ -42,3 +46,10 @@ func main() {
 	}
 
 }
+
+func formatBuiltWith(Tags string) string {
+	if len(Tags) > 0 {
+		return " built with: " + strings.Join(strings.Split(Tags, " "), ", ")
+	}
+	return ""
```
Comment (appleboy) [inline main.go]:
```go
 func formatBuiltWith(Tags string) string {
  	if len(Tags) == 0 {
  		return ""
  	}

  	return " built with: " + strings.Join(strings.Split(Tags, " "), ", ")
```

more readable?

Comment (lunny) [reply main.go]: Or `strings.Replace(Tags, " ", ", ", -1)`?

Comment (cez81) [reply main.go]: Yes, is guess the special case is empty tags. Will update with both suggestions.

**Comment by lunny** [comment]
It didn't work well. Because `Version` is also be set on the make progress. I think maybe you can only change the Makefile.
```
LDFLAGS := -X "main.Version=$(shell git describe --tags --always | sed 's/-/+/' | sed 's/^v//') $(TAGS)"
```
It's ok.

**Comment by cez81** [comment]
@lunny Don't think I understand where else you mean it's used. Can you explain please?

**Comment by tboerger** [comment]
I like how it's set now, but I'm not sure if I like how it's getting printed out

**Comment by tboerger** [comment]
The version also gets printed on the page itself. Maybe we want to print the tags also on the admin page

**Comment by lunny** [comment]
@cez81 like @tboerger said,  put it in admin panel maybe better.

File: main.go [hunk 2]
```diff
@@ -42,3 +46,11 @@ func main() {
...
 
 }
+
+func formatBuiltWith(Tags string) string {
+	if len(Tags) == 0 {
+  		return ""
+  	}
+
+  	return " built with: " + strings.Replace(Tags, " ", ", ", -1)
+}
```
Comment (tboerger) [inline main.go]: strings.Join

**Comment by tboerger** [comment]
I'm fine with displaying it as a comma separated list without quotes if I look on the referenced issue. But additionally it should be displayed on the admin ui.

File: main.go (hunk 2 above)
Comment (lunny) [reply main.go]: Tags is `sqlite bindata` form. Why use strings.Join ?

Comment (tboerger) [reply main.go]: True, but maybe it's even easier to create a comma separated list to split by space and join by comma

**Comment by cez81** [comment]
tags_admin_panel
tags_console

File: main.go [hunk 3]
```diff
@@ -18,15 +19,18 @@ import (
 // Version holds the current Gitea version
 var Version = "1.1.0+dev"
 
+// Tags holds the build tags used
+var Tags = ""
+
 func init() {
-	setting.AppVer = Version
+	setting.AppVer = Version + formatBuiltWith(Tags)
```
Comment (tboerger) [inline main.go]: This should be done with a separate variable, IMHO we should not list the build tags on every page in the footer, only within the admin dashboard

Comment (cez81) [reply main.go]: True, I missed that!

**Comment by lunny** [comment]
build failed.

File: modules/templates/helper.go [hunk 4]
```diff
@@ -48,6 +48,9 @@ func NewFuncMap() []template.FuncMap {
 		"AppVer": func() string {
 			return setting.AppVer
 		},
+                "AppBuiltWith": func() string {
+                        return setting.AppBuiltWith
+                },
```
Comment (appleboy) [inline modules/templates/helper.go]: coding style.

**Comment by tboerger** [comment]
LGTM

**Pull request merged**